axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchGenerateMoves.java

```diff
@@ -0,0 +1,435 @@
+      Map<Tax, BigDecimal> taxMap,
+      Map<PaymentMode, BigDecimal> paymentMap)
+      throws AxelorException {
+
+    Move move =
+        moveCreateService.createMove(
+            journal,
+            company,
+            partner,
+            currency,
+            period,
+            date,
+            tradingName,
+            null,
+            MoveRepository.FUNCTIONAL_ORIGIN_PURCHASE,
+            null,
+            journal.getDescriptionModel());
+
+    int ref = 1;
+    BigDecimal total = BigDecimal.ZERO;
+
+    for (Entry<Account, BigDecimal> entry : amountMap.entrySet()) {
+      ref = generateAccountMoveLine(date, move, ref, entry);
+      total = total.add(entry.getValue());
+    }
+
+    for (Entry<Tax, BigDecimal> entry : taxMap.entrySet()) {
+      ref = generateTaxMoveLine(company, date, move, ref, entry);
+      total = total.add(entry.getValue());
+    }
+
+    MoveLine debitMoveLine = generateDebitMoveLine(account, date, move, ref, total);
+
+    moveRepository.save(move);
+
+    Map<PaymentMode, Move> paymentMoveMap = new HashMap<>();
+
+    for (Entry<PaymentMode, BigDecimal> entry : paymentMap.entrySet()) {
+      Move paymentMove =
+          generatePaymentMove(
+              tradingName,
+              period,
+              company,
+              partner,
+              currency,
+              account,
+              date,
+              move,
+              debitMoveLine,
+              entry);
+      paymentMoveMap.put(entry.getKey(), paymentMove);
+    }
+
+    for (Invoice invoice : invoiceList) {
+      ventilateInvoice(move, invoice, accountConfig);
+      generateInvoicePayment(date, paymentMoveMap, invoice);
+    }
+
+    String origin =
+        invoiceList.stream().map(Invoice::getInvoiceId).collect(Collectors.toList()).toString();
+
+    move.setOrigin(origin);
+
+    Boolean dayBookMode =
+        accountConfigService.getAccountConfig(move.getCompany()).getAccountingDaybook()
+            && move.getJournal().getAllowAccountingDaybook();
+
+    moveValidateService.updateValidateStatus(move, dayBookMode);
+  }
+
+  @Transactional
+  protected InvoicePayment generateInvoicePayment(
+      LocalDate date, Map<PaymentMode, Move> paymentMoveMap, Invoice invoice) {
+    InvoicePayment invoicePayment = new InvoicePayment();
+    invoicePayment.setAmount(invoice.getInTaxTotal());
+    invoicePayment.setCurrency(invoice.getCurrency());
+    invoicePayment.setInvoice(invoice);
+    invoicePayment.setTypeSelect(InvoicePaymentRepository.TYPE_PAYMENT);
+    invoicePayment.setPaymentMode(invoice.getPaymentMode());
+    invoicePayment.setMove(paymentMoveMap.get(invoice.getPaymentMode()));
+    invoicePayment.setPaymentDate(date);
+    invoicePayment.setStatusSelect(InvoicePaymentRepository.STATUS_VALIDATED);
+    return invoicePaymentRepository.save(invoicePayment);
+  }
+
+  private void ventilateInvoice(Move move, Invoice invoice, AccountConfig accountConfig)
+      throws AxelorException {
```

Comment:
I don't think we can ventilate the invoice like this. We should call the existing process in order to don't avoid to forget a part of the process.